widgets/accordion/js/accordion.js

```diff
@@ -64,18 +69,34 @@ jQuery( function ( $ ) {
 				} else {
 					openPanel( $panel.get( 0 ) );
 				}
+
 				if ( ! isNaN( maxOpenPanels ) && maxOpenPanels > 0 && openPanels.length > maxOpenPanels ) {
-					closePanel( openPanels[ 0 ] );
+					skippedPanels = 0;
+					$.each( openPanels.reverse(), function( index, el ) {
+						if ( skippedPanels != maxOpenPanels) {
+							skippedPanels++;
+						} else {
+							closePanel( openPanels[ index ] );
+						}
+					} );
 				}
 			} );
 			
 			if ( useAnchorTags ) {
 				updateHash = function () {
 					var anchors = [];
-					for ( var i = 0; i < openPanels.length; i++ ) {
-						var anchor = $( openPanels[ i ] ).data( 'anchor' );
+					allOpenPanels = $( '.sow-accordion-panel-open' ).toArray();
+					for ( var i = 0; i < allOpenPanels.length; i++ ) {
+						var anchor = $( allOpenPanels[ i ] ).data( 'anchor' );
 						if ( anchor ) {
-							anchors[ i ] = anchor;
+							$parentPanel = $( allOpenPanels[ i ] ).parents( '.sow-accordion-panel' );
+							if ( $parentPanel.length ) {
```

Comment:
Both code paths are doing the same thing, so this can just be a single condition. I.e. something like 
`if( ! $parentPanel.length || ( $parentPanel.length && $parentPanel.hasClass( 'sow-accordion-panel-open' ) ) )`